runtime/tests/port/si_numcpusTest.c

```diff
@@ -116,17 +116,45 @@ int j9sysinfo_numcpus_test5 (J9PortLibrary* portLibrary)
 	return reportTestExit(portLibrary, testName);
 }
 
+/**
+ * Set user-specified to 10. get target number of CPUs. Validate number is 10.
+ *
+ * @param[in] portLibrary The port library under test
+ *
+ * @return TEST_PASSED on success, TEST_FAILED on failure
+ */
+int j9sysinfo_numcpus_test4 (J9PortLibrary* portLibrary)
+{
+#define J9SYSINFO_NUMCPUS_TEST6_SPECIFIED 10
+	PORT_ACCESS_FROM_PORT(portLibrary);
+	const char* testName = "j9sysinfo_numcpus_test4";
+	UDATA targetNumberCPUs = 0;
+
+	j9sysinfo_set_number_user_specified_CPUs(J9SYSINFO_NUMCPUS_TEST6_SPECIFIED);
+	targetNumberCPUs = j9sysinfo_get_number_CPUs_by_type(J9PORT_CPU_TARGET);
+
+	outputComment(PORTLIB,"\nSet number of entitled CPUs to %d. Get target number of CPUs.\n", J9SYSINFO_NUMCPUS_TEST6_SPECIFIED);
+	outputComment(PORTLIB,"	Expected: %d\n", J9SYSINFO_NUMCPUS_TEST6_SPECIFIED);
+	outputComment(PORTLIB,"	Result:   %d\n", targetNumberCPUs);
+
+	if (J9SYSINFO_NUMCPUS_TEST6_SPECIFIED != targetNumberCPUs) {
```

Comment:
There should also be a test that sets active cpus to zero, after setting something else, and confirms the target value goes back to the original value.
Plus a test that confirms a negative value cannot be set.

Reply:
Right now, a negative value can technically be set (omr does not throw an error in `j9sysinfo_set_number_user_specified_CPUs`), but it would have the same effect as setting 0 (active CPUs are then not considered).

But since the command line option cannot parse negative numbers, and negative numbers are not accepted by the ProcessorMXBean function for consistency, a negative value effectively cannot be set (there just isn't a test to be written here). I think this is fine.

Reply:
Yes I agree. I didn't realize at first that omrsysinfo_get_number_CPUs_by_type takes an unsigned value.